test/integration_test/utils/storagecluster.go

```diff
@@ -34,17 +35,58 @@ func CreateStorageClusterTestSpecFunc(cluster *corev1.StorageCluster) func(t *te
 
 // ConstructStorageCluster makes StorageCluster object and add all the common basic parameters that all StorageCluster should have
 func ConstructStorageCluster(cluster *corev1.StorageCluster, specGenURL string, specImages map[string]string) error {
+	var envVarList []v1.EnvVar
+
 	cluster.Spec.Image = specImages["version"]
 	if cluster.Namespace == "" {
 		cluster.Namespace = PxNamespace
 	}
 
+	if cluster.Namespace != PxNamespace {
+		// Create custom namespace
```

Comment:
IMO we should avoid creating this ns here, it should be created in the test itself, otherwise how can we make sure it's deleted after the test? if not deleted properly it could cause other test failures.

Reply:
Done, moved it to test